addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/device/DimmerDevice.java

```diff
@@ -38,24 +38,36 @@
     @Override
     public boolean handleCommand(String channelId, Connection connection, Command command,
             TPLinkSmartHomeConfiguration configuration) throws IOException {
-        return CHANNEL_BRIGHTNESS.equals(channelId) ? handleBrightness(channelId, connection, command, configuration)
+        return CHANNEL_BRIGHTNESS.equals(channelId)
+                ? handleBrightnessChannel(channelId, connection, command, configuration)
                 : super.handleCommand(channelId, connection, command, configuration);
     }
 
-    private boolean handleBrightness(String channelId, Connection connection, Command command,
+    /**
+     * Handle the brightness channel. Because the device has different commands for setting the device on/off and
+     * setting the brightness the on/off command must be send to the device as well when the brightness.
+     *
+     * @param connection Connection to use
+     * @param command command to the send
+     * @return returns true if the command was handled
+     * @throws IOException throws an {@link IOException} if the command handling failed
+     */
+    private boolean handleBrightnessChannel(String channelId, Connection connection, Command command,
             TPLinkSmartHomeConfiguration configuration) throws IOException {
         HasErrorResponse response = null;
 
         if (command instanceof OnOffType) {
             response = setOnOffState(connection, (OnOffType) command);
-        } else if (command instanceof DecimalType) {
-            DecimalType decimalCommand = (DecimalType) command;
+        } else if (command instanceof PercentType) {
```

Comment:
Is this a breaking change?

Reply:
No, the channel is a `Dimmer`, so it should have been `PercentType` all along. It worked because `DecimalType` is a superclass so it won't complain here.